test/Microsoft.Dnx.ApplicationHost.FunctionalTests/DllImportTests.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.Extensions.PlatformAbstractions;
+using Microsoft.Dnx.Testing.Framework;
+using Xunit;
+
+namespace Microsoft.Dnx.ApplicationHost.FunctionalTests
+{
+    public class DllImportTests : DnxSdkFunctionalTestBase
+    {
+        [Theory, TraceTest]
+        [MemberData(nameof(DnxSdks))]
+        public void VerifyDllImportWithProjectReference(DnxSdk sdk)
+        {
+            VerifyDllImport(sdk, "ProjectReferenceTest");
+        }
+
+        [Theory, TraceTest]
+        [MemberData(nameof(DnxSdks))]
+        public void VerifyDllImportWithPackageReference(DnxSdk sdk)
+        {
+            VerifyDllImport(sdk, "PackageReferenceTest");
+        }
+
+        public void VerifyDllImport(DnxSdk sdk, string projectName)
+        {
+            var runtimeEnvironment = PlatformServices.Default.Runtime;
+
+            // Preloading .so's on Linux does not work and setting
+            // LD_LIBRARY_PATH is not an option so skipping
+            if (runtimeEnvironment.OperatingSystem == "Linux" && runtimeEnvironment.RuntimeType == "Mono")
```

Comment:
We suck - the skip attributes are independent so Linux and Mono will skip all Linux and all Mono and not Mono on Linux. Leaving as is.